raft/src/main/java/org/apache/kafka/snapshot/RecordsSnapshotWriter.java

```diff
@@ -216,4 +140,132 @@ private void appendBatches(List<CompletedBatch<T>> batches) {
             batches.forEach(CompletedBatch::release);
         }
     }
+
+    final public static class Builder {
+        private long lastContainedLogTimestamp = 0;
+        private CompressionType compressionType = CompressionType.NONE;
+        private Time time = Time.SYSTEM;
+        private int maxBatchSize = 1024;
+        private MemoryPool memoryPool = MemoryPool.NONE;
+        private short kraftVersion = 0;
```

Comment:
Why would we want most tests to use an older version?

Reply:
Good point. I made it kraft version 0 because most tests today are for that version. I'll change the default to 1 since we want new tests to at least target kraft version 1.